Standards/CCSMB-3.md

```diff
@@ -0,0 +1,109 @@
+# CCSMB-3 bimg File Format
+
+*Author: SkyTheCodeMaster*
+
+*Contributers: MasonGulu, spr2-dev*
+
+*Version: 1.0.0*
+
+*Last Revised: 2022-07-29*
+
+bimg (abbreviation of "blit image") is a versatile image format for the ComputerCraft Minecraft mod.
+This specification aims to provide a standardized image format that can be easily implemented and used throughout the entire ComputerCraft ecosystem.
+
+# File Structure
+The examples shown throughout this specification are based on a sample image file, serialized using `textutils.serialize`.
+
+An image file can be separated into two sections: metadata and frame data. These two sections are unordered, meaning that they can be correctly written and read by parsers regardless of whether the metadata or the frame data comes first:
+![Format overview](media/CCSMB-3/file-structure.png?raw=true)
+
+Metadata contains information about the image itself, while the remaining of the file contains one or more frames, representing the images.
+
+For more information on metadata, see [Metadata](#metadata). For more information on the image data and frames, see [Image Data](#image-data)
+.
+# Metadata
+bimg stores the following metadata along the image data in the file.
+
+| Field | Type | Description |
+|:--|:-:|:--|
+| `version` | `string` | Format version of the file. Used to differentiate two bimg file formats. Composed of three numbers separated by dots, defining the major, minor, and revision number respectively. |
```

Comment:
It would be nice if this value included the string "BIMG", so that the file format can be determined immediately without needing to probe the rest of the file.

Reply:
i think this should be a seperate field, to ensure backwards compatibility